osf/models/osf_group.py

```diff
@@ -0,0 +1,244 @@
+from django.apps import apps
+from django.db import models
+from django.core.exceptions import ValidationError
+from guardian.shortcuts import assign_perm, remove_perm, get_perms
+from guardian.models import GroupObjectPermissionBase, UserObjectPermissionBase
+from framework.exceptions import PermissionsError
+from framework.auth.core import get_user
+
+from osf.models import base
+from osf.models.mixins import GuardianMixin
+from osf.models.user import OSFUser
+from osf.utils.permissions import ADMIN
+from osf.utils import sanitize
+
+MEMBER = 'member'
+MANAGER = 'manager'
+
+# TODO Add logging for OSFGroup actions
+# TODO Send email to member/unregistered member when added to group
+# TODO OSFGroups should either have a guid or a longer _id, assuming guid for now, can remove
+
+
+class OSFGroup(GuardianMixin, base.GuidMixin, base.BaseModel):
```

Comment:
Initial modeling for OSFGroups.   Like AbstractNode, the OSFGroup model uses the Guardian mixin.  On creation, a "member" and "manager" django group are created for it.  Managers have manage permissions.  Membership in the manager group allows you to add/update/remove other managers/members from the OSF Group.  All OSF group members belong the member group.  This member group as a whole can be given permissions to othe rnodes.